### ISSUE TITLE
review bin/fetch scripts to try mirrors when first one fails instead of retryning same mirror

### ISSUE THREAD

**Issue by tlaurion** [linuxboot/heads#1610]
CircleCI showed an issue when attempting to build https://app.circleci.com/pipelines/github/tlaurion/heads?branch=pre423-edp_fhd_in-tpm_tcg_event_log-CircleCi_coreboot_cache_fixed

See mpfr download failing here: https://app.circleci.com/pipelines/github/tlaurion/heads/2275/workflows/f6d1bfb3-f5f5-4141-a8d6-5afa95a2ce0a/jobs/40709?invite=true#step-103-4141_69

**Comment by JonathonHall-Purism** [comment]
TL;DR: bin/fetch_coreboot_crossgcc_archive.sh might want to try the wget more than once, maybe only if the remote host is ftpmirror.gnu.org, or maybe try it again after checking our mirrors.

This has to do with packages from `ftpmirror.gnu.org`.  It redirects to different mirrors when we try to fetch from it.

In that log, it redirected to `mirrors.ocf.berkeley.edu` but it seems we were unable to reach that host.  wget retries internally, but only from the redirected address (meaning it kept failing on the same mirror), it does not go back to the original address to see if the redirect would be different (which would be a bit silly outside of cases like this).

If we tried the wget again, there's a good chance we would get a different mirror, which would shield us from outages of an individual mirror.

This branch also had new crossgcc dependencies that aren't in the mirror on storage.puri.sm yet, so our mirrors didn't help.
